### DISCUSSION TITLE
[React] Preserve auth data between sessions?

### DISCUSSION THREAD

**Discussion by zfei** [fief-dev/fief#141] [Q&A] [answered]
It looks like the browser `FiefAuthStorage` sync's auth data to session storage whereas `FiefReactAuthStorage` only sync's them to a state (reducer).

Currently with client-side routing, the auth info works fine, but once I refresh the page, the state's gone. How do we preserve `userinfo`/`tokeninfo` between sessions then?

**Comment by fief-bailiff** [comment]
Hail, @zfei 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Hello @zfei 👋 

That's indeed a limitation of our React implementation, and maybe we should revisit this to store it also in storage. I'll add this to the backlog.

That said, when accessing a private page, the user shall be redirected to Fief's login page and, as we store a session on our side, they should be automatically redirected to your app again. So, this should be quite fast and transparent for them.

**Reply by zfei** [reply]
Thanks for the reply. Thing is the UX could be a bit abrupt here. In my case, it's a section of the page that's protected, instead of the entire page. When the user's not authenticated, the protected section would just render a banner suggesting the user to sign up/login if they wish to.

**Reply by frankie567** [reply]
Yeah, I can see the problem. I'll look at this ASAP!

**Comment by frankie567** [comment, marked as answer]
This shall be solved in the latest version of the JS SDK: https://github.com/fief-dev/fief-js/releases/tag/v0.12.1

**Reply by zfei** [reply]
This is awesome, thanks!
